src/ComplexType.php

```diff
@@ -273,27 +333,20 @@ public function getMembers()
     /**
      * Generate a string representing the parameters for a function e.g. "type1 $param1, type2 $param2, $param3"
      *
-     * @param array $parameters A map of parameters. Keys are parameter names and values are parameter types.
-     *                          Parameter types may be empty. In that case they are not used.
-     * @param bool $includeType Whether to include the parameters types in the string
-     * @param bool $defaultNull Whether to set the default value of parameters to null.
+     * @param ParameterGenerator[] $parameters A map of parameters
      * @return string The parameter string.
      */
-    protected function buildParametersString(array $parameters, $includeType = true, $defaultNull = false)
+    protected static function buildParametersString(array $parameters)
     {
-        $parameterStrings = array();
-        foreach ($parameters as $name => $type) {
-            $parameterString = '$' . $name;
-            if (!empty($type) && $includeType) {
-                $parameterString = $type . ' ' . $parameterString;
-            }
-            if ($defaultNull) {
-                $parameterString .= ' = null';
+        $parameterOutput = [];
+
+        if (!empty($parameters)) {
```

Comment:
Since we typehint `$parameters` to be an array there is no reason to check for empty.